src/couch_expiring_cache/src/couch_expiring_cache.erl

```diff
@@ -0,0 +1,33 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_expiring_cache).
+
+-export([
+    lookup/2,
+    insert/5
+]).
+
+
+-include_lib("couch_expiring_cache/include/couch_expiring_cache.hrl").
+
+
+-spec insert(Name :: binary(), Key :: binary(), Value :: binary(),
+    StaleTS :: millisecond(), ExpiresTS :: millisecond()) -> ok.
+insert(Name, Key, Value, StaleTS, ExpiresTS) ->
```

Comment:
Wonder if we could update so that we can pass an optional transaction in. That is, user would insert and lookup values inside another transaction which is already started. 

This would complicate the picture a bit, not sure if it is worth it...

For example take a look at couch_jobs:

https://github.com/apache/couchdb/blob/prototype/fdb-layer/src/couch_jobs/src/couch_jobs.erl#L63-L70

There Tx could be undefined, not transaction passed in. It could be a `naked` erlfdb Tx object or a fabric2_db database handle `#{tx := , layer_prefix := ...}` object.  Some of that logic is handled here:

https://github.com/apache/couchdb/blob/prototype/fdb-layer/src/couch_jobs/src/couch_jobs_fdb.erl#L412-L439

Reply:
https://github.com/apache/couchdb/pull/2331/commits/b9392164bf6f86dc3f2906959c9226ab55e6e347 leverages `couch_jobs_fdb` to optionally allow `insert` and `lookup` to occur inside an existing transaction. The main drawback seems to be that key calculation now always occurs inside the transaction, whereas before in was outside.

Reply:
nice work!